pkg/relayer/proxy/synchronous.go

```diff
@@ -249,14 +250,28 @@ func (sync *synchronousRPCServer) serveHTTP(
 		return nil, err
 	}
 
+	// Check if the response is gzip-compressed
+	// TODO: Add other algorithms, or an alternative would be to switch to http
+	// client that manages all low-level HTTP decisions for us, something like
+	// https://github.com/imroc/req, https://github.com/valyala/fasthttp or
+	// https://github.com/go-resty/resty
+	var responseBody io.ReadCloser = httpResponse.Body
+	if httpResponse.Header.Get("Content-Encoding") == "gzip" {
+		responseBody, err = gzip.NewReader(httpResponse.Body)
```

Comment:
Can you create a helper that takes `httReponse` and returns the correct one? It'll be easier to add new ones in the future.

Reply:
Great idea!